sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingSymmetricHashJoinExec.scala

```diff
@@ -637,18 +653,22 @@ case class StreamingSymmetricHashJoinExec(
         thisRow: UnsafeRow,
         subIter: Iterator[InternalRow])
       extends CompletionIterator[InternalRow, Iterator[InternalRow]](subIter) {
-
+      // scalastyle:off
       private val iteratorNotEmpty: Boolean = super.hasNext
 
       override def completion(): Unit = {
         val isLeftSemiWithMatch =
           joinType == LeftSemi && joinSide == LeftSide && iteratorNotEmpty
         // Add to state store only if both removal predicates do not match,
         // and the row is not matched for left side of left semi join.
+        println(s"!stateKeyWatermarkPredicateFunc(key): ${!stateKeyWatermarkPredicateFunc(key)}" +
+          s" !stateValueWatermarkPredicateFunc(thisRow): ${!stateValueWatermarkPredicateFunc(thisRow)}")
         val shouldAddToState =
           !stateKeyWatermarkPredicateFunc(key) && !stateValueWatermarkPredicateFunc(thisRow) &&
           !isLeftSemiWithMatch
         if (shouldAddToState) {
+          println(s"wei==add to state: $thisRow")
```

Comment:
If comment out the if (shoudAddToState) above, then there is data:
```
[info] - SPARK-45637 window agg + window agg -> join on window, append mode *** FAILED *** (6 seconds, 953 milliseconds)
[info]   == Results ==
[info]   !== Correct Answer - 1 ==              == Spark Answer - 1 ==
[info]   !struct<_1:int,_2:int,_3:int,_4:int>   struct<window:struct<start:timestamp,end:timestamp>,count:bigint,count:bigint>
[info]   ![0,5,5,1]                             [[1969-12-31 16:00:00.0,1969-12-31 16:00:05.0],1,5]
[info] 
```
My [0, 5, 5, 1] answer is badly written but the idea is the same as the spark answer

Reply:
So what happens here is in the no data batch, the wm of `stateKeyWatermarkPredicateFunc` is updated to the new global wm (8), the emitted key from both parent window aggregations are [0, 5). Hence `stateKeyWatermarkPredicateFunc(key)` returns true, meaning that the window should not be added to the join state store.

This is wrong, because the two [0, 5) windows should be joined here. 

This looks like some updates to the multiple state operators that we need to consider

Reply:
stateKeyWatermarkPredicateFunc's wm should be 0 here, not 8

Reply:
https://github.com/apache/spark/commit/991726f31a8d182ed6d5b0e59185d97c0c5c532f